admin/services/pupil.service.js

```diff
@@ -87,6 +87,28 @@ const pupilService = {
       }
     }
     return true
+  },
+  /**
+   * Adds show Date of Birth flag for pupils with equal full names
```

Comment:
There should be a note to say this *assumes* the array of objects have been sorted by some criteria.